### PR TITLE
feat(outputs.websocket): Allow specifying secrets in headers

### LINKED ISSUE #14835: feat(outputs.websocket): support secrets in headers
### Use Case

I want to livestream data to Grafana but a custom Authorization header is needed for that.

### Expected behavior

Data streams to Grafana instead of packet based with the http.output plugin. The secret can be set without leaking it.

### Actual behavior

Not supported

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by LarsStegman**
## Summary
Make the header values secrets instead of plain strings to support setting bearer tokens properly.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves #14835

### CODE REVIEW

File: plugins/outputs/websocket/websocket.go [hunk 1]
```diff
@@ -92,7 +92,14 @@ func (w *WebSocket) Connect() error {
 
 	headers := http.Header{}
 	for k, v := range w.Headers {
-		headers.Set(k, v)
+		secret, err := v.Get()
+		if err != nil {
+			return err
```
Comment (srebhan) [inline plugins/outputs/websocket/websocket.go:97]: Please ease debugging by giving a hint which header failed
```suggestion
			return fmt.Errorf("getting header secret %q failed: %w", k, err)
```

File: plugins/outputs/websocket/websocket.go [hunk 2]
```diff
@@ -92,7 +92,14 @@ func (w *WebSocket) Connect() error {
...
+		}
+
+		headerVal := secret.String()
+		headers.Set(k, headerVal)
```
Comment (srebhan) [inline plugins/outputs/websocket/websocket.go:101]: Purely cosmetic but I think this looks better:
```suggestion
		headers.Set(k, secret.String())
```

**Review by srebhan — COMMENTED** [review]
@LarsStegman thanks for the PR! Just two mostly cosmetic comments...

**Review by srebhan — APPROVED** [review]
Thanks @LarsStegman!

**Review by powersj — APPROVED** [review]

**Pull request merged**